tests/test_main.py

```diff
@@ -1,11 +1,209 @@
+    ]
+    assert data == expected_data
+
+
+def test_get_project_details(client: TestClient, created_project: ProjectDetails) -> None:
+    response = client.get(f"/project/{created_project.project_id}/info")
+    assert response.status_code == 200
+    data = response.json()
+    expected_data = {
+        "name": created_project.name,
+        "description": created_project.description,
+        "project_id": str(created_project.project_id),
+    }
+    assert data == expected_data
 
-client = TestClient(app)
 
+def test_get_project_details_not_found(client: TestClient) -> None:
+    random_uuid = uuid.uuid4()
+    response = client.get(f"/project/{random_uuid}/info")
+    assert response.status_code == 404
+    assert response.json() == {"detail": f"ProjectDetails {random_uuid} not found"}
 
-def test_read_root() -> None:
-    response = client.get("/")
+
+def test_update_project_details(client: TestClient, created_project: ProjectDetails) -> None:
+    project = ProjectDetails(
+        project_id=created_project.project_id, name=created_project.name, description=created_project.description
+    )
+    projects.append(project)
+    retrieved_project = get_project(created_project.project_id)
+    assert retrieved_project == project
+
+    updated_data = {"name": "Updated Name", "description": "Updated project description"}
+    response = client.put(f"/project/{created_project.project_id}/info", json=updated_data)
     assert response.status_code == 200
-    assert response.json() == {"message": "Hello World"}
+    data = response.json()
+    expected_data = {
+        "name": created_project.name,
+        "description": created_project.description,
+        "project_id": str(created_project.project_id),
+    }
+    assert data == expected_data
+
+
+def test_update_project_details_not_found(client: TestClient) -> None:
+    random_uuid = uuid.uuid4()
+    updated_data = {"name": "Updated Name", "description": "Updated project description"}
+    response = client.put(f"/project/{random_uuid}/info", json=updated_data)
+    assert response.status_code == 404
+    assert response.json() == {"detail": f"ProjectDetails {random_uuid} not found"}
+
+
+def test_delete_project(client: TestClient, created_project: ProjectDetails) -> None:
+    response = client.delete(f"/project/{created_project.project_id}")
+    assert response.status_code == 204
+    assert response.content == b""
+
+
+def test_delete_project_not_found(client: TestClient) -> None:
+    random_uuid = uuid.uuid4()
+    response = client.delete(f"/project/{random_uuid}")
+    assert response.status_code == 404
+    assert response.json() == {"detail": f"ProjectDetails {random_uuid} not found"}
+
+
+def test_create_project_service() -> None:
+    project = Project(name="Test Project", description="Test Description")
+    created_project = create_project_service(project)
+    assert created_project.name == "Test Project"
+    assert created_project.description == "Test Description"
```

Comment:
Check the whole output - that it is the data type you want and also all other fields. Do not cherry pick them